gui/wxpython/gui_core/infobar.py

```diff
@@ -0,0 +1,146 @@
+
+import wx
+import wx.aui
+try:
+    import wx.lib.agw.infobar as IB
+except ImportError:
+    import wx.lib.infobar as IB
+
+from grass.script import gisenv
+
+
+class InfoBar(IB.InfoBar):
+    """
+    Custom Info bar
+    """
+    def __init__(self, parent):
+        IB.InfoBar.__init__(self, parent)
+
+        self.SetBackgroundColour(wx.Colour(255, 248, 220))
+        self._text.SetOwnForegroundColour(wx.SystemSettings.GetColour(wx.SYS_COLOUR_GRAYTEXT))
+
+        self.buttons_ids = []
+
+        # LAYOUT
+        # icon, text, close button
+        sizer = wx.BoxSizer(wx.HORIZONTAL)
+        self.buttonBoxSizer = wx.BoxSizer(wx.VERTICAL)
+        self.buttonBoxSizer.Add(self._button, wx.SizerFlags().Top().Border(wx.ALL, 5))
+
+        sizer.Add(self._icon, wx.SizerFlags().Top().Border(wx.ALL, 5))
+        sizer.Add(self._text, proportion=1,
+                  flag=wx.EXPAND | wx.ALL, border=5)
+        sizer.Add(self.buttonBoxSizer, wx.SizerFlags().Top().Border(wx.ALL, 5))
+
+        self.SetSizer(sizer)
+
+    def AddButton(self, btnid, label, bitmap=wx.NullBitmap):
+        """
+        Adds a button to be shown in the info bar and
+        does not remove Close button.
+        """
+        sizer = self.GetSizer()
+
+        if not sizer:
+            raise Exception("must be created first")
+
+        button = wx.Button(self, btnid, label)
+
+        # Add another button under close button
+        self.buttonBoxSizer.Add(button, wx.SizerFlags().Top().Border(wx.ALL, 5))
+
+        if self.IsShown():
+            self.UpdateParent()
+
+        return button
+
+    def CreateButton(self, button_dict):
```

Comment:
This shouldn't be dictionary, order in dictionary is not defined, should be something like list of tuples.